packages/apollo-server-lambda/src/lambdaApollo.ts

```diff
@@ -51,10 +51,16 @@ export function graphqlLambda(
       },
     }).then(
       ({ graphqlResponse, responseInit }) => {
+        let graphqlResponseJson = JSON.parse(graphqlResponse);
+        if(graphqlResponseJson.errors != undefined) {
+          graphqlResponseJson.code = "error";
```

Comment:
why are we setting `code`? doesn't API gateway look for `errorMessage` field? the normal call is `callback(null, {body, statusCode, headers})`. Your modified call is just `callback(graphqlResponse)`. how does that work?

also, we are checking for any errors at all. are there different types or errors? maybe we only want to fail for certain kinds of errors?

in any case. do we log these errors currently? if not, we should probably do that here as well

Reply:
what if instead of hardcoding our app specific code here, we create a function in `ApolloServer.ts` called `setResponseInterceptor` which takes a function the gets passed body, statusCode, headers and returns an object with the resultant body, statusCode, and headers?

if it's unset we do nothing here. and if that function gets set, then we call it here and perform `callback` with the modified, body code and headers. that way we can keep our app specific code out of the graphql code